test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -178,13 +183,27 @@ public void initValuesSourceRegistry() {
         valuesSourceRegistry = searchModule.getValuesSourceRegistry();
     }
 
+    @Before
+    public void initAnalysisRegistry() throws IOException {
+        analysisModule = new AnalysisModule(
+            TestEnvironment.newEnvironment(
+                Settings.builder().put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString()).build()
```

Comment:
I'd sort of hoped not to pull in the whole world in these tests. I feel like making a tmp dir is crossing some kind of line. I dunno the right thing. Maybe mockito for the `Environment`?